buildenv/Build_Instructions_V9.md

```diff
@@ -0,0 +1,375 @@
+```
+
+  - For Linux on 64-bit Power systems, run:
+```
+wget https://raw.githubusercontent.com/eclipse/openj9/master/buildenv/docker/jdk9/ppc64le/ubuntu16/Dockerfile
+```
+
+  - For Linux on 64-bit z Systems, run:
+```
+wget https://raw.githubusercontent.com/eclipse/openj9/master/buildenv/docker/jdk9/s390x/ubuntu16/Dockerfile
+```
+
+3. Next, run the following command to build a Docker image, called **openj9**:
+```
+docker build -t openj9 Dockerfile .
+```
+
+4. Start a Docker container from the **openj9** image with the following command, where `-v` maps any directory, `<host_directory>`,
+on your local system to the containers `/root/hostdir` directory so that you can store the binaries, once they are built:
+```
+docker run -v <host_directory>:/root/hostdir -it openj9
+```
+
+:pencil: Depending on your [Docker system configuration](https://docs.docker.com/engine/reference/commandline/cli/#description), you might need to prefix the `docker` commands with `sudo`.
+
+Now that you have the Docker image running, you are ready to move to the next step, **Get the source**.
+
+#### Setting up your build environment without Docker
+
+If you don't want to user Docker, you can still build an OpenJDK V9 with OpenJ9 directly on your Ubuntu system or in a Ubuntu virtual machine. Use the
+[Linux on x86 Dockerfile](https://github.com/eclipse/openj9/blob/master/buildenv/docker/jdk9/x86_64/ubuntu16/Dockerfile) like a recipe card to determine the software dependencies
+that must be installed on the system, plus a few configuration steps.
+
+:pencil:
+Not on x86? We also have Dockerfiles for the following Linux architectures: [Linux on Power systems](https://github.com/eclipse/openj9/blob/master/buildenv/docker/jdk9/ppc64le/ubuntu16/Dockerfile) and [Linux on z Systems](https://github.com/eclipse/openj9/blob/master/buildenv/docker/jdk9/s390x/ubuntu16/Dockerfile).
+
+
+1. Install the dependencies that can be obtained with the `apt-get` command. Copy
+and paste the following command section from the Dockerfile to your command prompt.
+```
+apt-get update \
+  && apt-get install -qq -y --no-install-recommends \
+    autoconf \
+    ca-certificates \
+    ...
+    ...
+    ...
+    ...
+  && rm -rf /var/lib/apt/lists/*
+```
+
+:pencil: For Linux on z Systems, we specify the [IBM SDK for Java 8](https://developer.ibm.com/javasdk/downloads/sdk8/) in the Dockerfile rather than the `openjdk-8-jdk` package because the IBM version contain a JIT compiler that will significantly accelerate compile time.
+
+2. This build uses the same gcc and g++ compiler levels as OpenJDK, which might be
+backlevel compared with the versions you use on your system. Create links for
+the compilers with the following command:
+```
+ln -s g++ /usr/bin/c++ \
+  && ln -s g++-4.8 /usr/bin/g++ \
+  && ln -s gcc /usr/bin/cc \
+  && ln -s gcc-4.8 /usr/bin/gcc
+```
+
+3. Download and setup **freemarker.jar** into the /root directory:
```

Comment:
We should say something like:
``` Download and setup **freemarker.jar** into a directory.  The example commands use `/root` to be consistent with the Docker instructions.  If you aren't using Docker, it's better to store the freemarker.jar in your home directory.```

As putting files in /root isn't a great practice.

Reply:
Agreed. I've made these changes.